watch.go

```diff
@@ -103,48 +121,54 @@ func (fb *Firebase) Watch(notifications chan Event) error {
 			// 		event: put
 			// 		data: {"path":"/","data":{"foo":"bar"}}
 
-			var evt []byte
-			var dat []byte
-			isPrefix := true
-			var result []byte
-
-			// For possible results larger than 64 * 1024 bytes (MaxTokenSize)
-			// we need bufio#ReadLine()
-			// 1. step: scan for the 'event:' part. ReadLine() oes not return the \n
-			// so we have to add it to our result buffer.
-			evt, isPrefix, scanErr = scanner.ReadLine()
+			payload := make([]string, 2)
+			scanErr = func(payload []string) error {
```

Comment:
It's OK to make a local function. I can do this.

Suppose that the reply is something like this:

event: someEvent
data: { some JSON data }
*other-header: other-info*
<empty line>

I know that this is not supposed to occur, but I wanted to handle such
cases anyhow.

Additionally, as far as I understand your code, you are ignoring the first
line:

line, scanErr = scanner.ReadString('\n') <- first line read here
> for ; lineCount != len(payload); line, scanErr = scanner.ReadString('\n')
> { <- first line overwritten here


but perhaps I missed something in your code.

█  Jacques

On 31 July 2016 at 21:26, Steven Berlanga <notifications@github.com> wrote:

> In watch.go
> <https://github.com/zabawaba99/firego/pull/50#discussion_r72909323>:
>
> > @@ -103,48 +121,54 @@ func (fb *Firebase) Watch(notifications chan Event) error {
> >  			// 		event: put
> >  			// 		data: {"path":"/","data":{"foo":"bar"}}
> >
> > -			var evt []byte
> > -			var dat []byte
> > -			isPrefix := true
> > -			var result []byte
> > -
> > -			// For possible results larger than 64 * 1024 bytes (MaxTokenSize)
> > -			// we need bufio#ReadLine()
> > -			// 1. step: scan for the 'event:' part. ReadLine() oes not return the \n
> > -			// so we have to add it to our result buffer.
> > -			evt, isPrefix, scanErr = scanner.ReadLine()
> > +			payload := make([]string, 2)
> > +			scanErr = func(payload []string) error {
>
> It is indeed an option. The closure is for me a way to avoid the "break"
> with a label. which I don't really like; but this is probably a question of
> personal style.
>
> If the case is to avoid having to break a label, can you extract this into
> it's own function then?
>
> The other benefits of my solution is that it handles correctly the case
> where a reply has more than 2 lines. Perhaps this is what causes the
> confusion.
>
> Would you mind giving me an example of a string that would have more than
> two lines that would not be handled correctly by both pieces of code?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zabawaba99/firego/pull/50/files/91637d68c23bcd6761930b23882b18621b759365#r72909323>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAB2iJZzdirda6_jdfY_3tTU0x47GU8nks5qbPbfgaJpZM4JYHTS>
> .
>